Create custom mode for error messages

rgrinberg opened the pull request:
Error messages that are too long, are displayed in their own buffer. We
make sure that this buffer has its own mode. This makes it easy to
customize it with hooks and keybindings.

This addressed my issue in #1294

cpitclaudel commented on flycheck.el:
```diff
@@ -4862,9 +4866,13 @@ In the latter case, show messages in the buffer denoted by
+      (let ((result (display-message-or-buffer (string-join messages "\n\n")
+                                               flycheck-error-message-buffer
+                                               'not-this-window)))
+        (if (window-live-p result)
```
This could be a `when`

cpitclaudel commented on flycheck.el:
```diff
@@ -4862,9 +4866,13 @@ In the latter case, show messages in the buffer denoted by
+                                               'not-this-window)))
+        (if (window-live-p result)
+            (with-current-buffer (window-buffer result)
+              (when (not (eq major-mode 'flycheck-error-message-mode))
```
This could be an `unless`, and I think the convention is to use `derived-mode-p`

cpitclaudel commented on flycheck.el:
```diff
@@ -4849,6 +4849,10 @@ The echo area may be used if the cursor is not in the echo area,
 
+(define-derived-mode flycheck-error-message-mode fundamental-mode
+  "Flycheck error messages"
+  "Major mode for extended error messages")
```
Missing `.` at the end of the docstring.
Also, should the mode derive from text-mode?

rgrinberg commented:
> LGTM, with a few minor tweaks. Feel free to merge after that ^^

Thanks for the review. I addressed your comments, but I don't have merging rights. A maintainer will need to do that.
